File: BPG_CMS_BE/src/BPG.Domain/Entities/Unit.cs
namespace BPG.Domain.Entities;

public class Unit : BaseEntity
{
    public int UnitId { get; set; }
    public string UnitCode { get; set; } = string.Empty;
    public string UnitName { get; set; } = string.Empty;

    public ICollection<MaterialCatalog> MaterialsAsBaseUnit { get; set; } = new List<MaterialCatalog>();
    public ICollection<MaterialConversion> AlternativeConversions { get; set; } = new List<MaterialConversion>();
}

File: BPG_CMS_FE/src/types/unit.ts
export interface Unit {
  unitId: number;
  unitCode: string;
  unitName: string;
  isDiscrete: boolean;
}

export interface CreateUnitRequest {
  unitCode: string;
  unitName: string;
  isDiscrete: boolean;
}

export interface UpdateUnitRequest {
  unitCode: string;
  unitName: string;
  isDiscrete: boolean;
}
